ADD. 1. update user Info, show more information 2. modify Repos info with description and update timestamp 3. add another column Projects and show projects as list like Repos

ritata opened the pull request.

ritata committed:
initial commit

chucklien commented on src/App.js:
```diff
@@ -1,5 +1,6 @@
 import React from "react";
 import "./styles.css";
+import ConvertToDate from "./Util.js";
```
in react, we usually us `Captital` as component, and others use lower case.
so recommend use `convertToDate` and `util.js`

chucklien replied:
if you take `ConvertToDate` as a react component, you could ignore this.

ritata replied:
since I will modify it to pure js function, so I will use `convertToDate` and `util.js`.

chucklien commented on src/Util.js:
```diff
@@ -0,0 +1,4 @@
+export default function ConvertToDate(props) {
+  const date = new Date(props.time).toLocaleDateString();
+  return date;
+}
```
I thought it's util function to get local date string, not a  react-component.
as i know, for react component, we need return a react.node not a pure string, that's what I mention above, use lower case for `convertToDate`

chucklien replied:
since this is used to get locale date string, you could treat this as pure js function.

ritata replied:
ok, I will modify it to pure JS function.

chucklien commented on src/App.js:
```diff
@@ -10,31 +11,74 @@ function Profile(props) {
+  ));
+  return <ul>{listProjects}</ul>;
+}
+
```
please check this two component `FavoriteRepos` and `MyProjects`, don't you think they look familiar?

chucklien replied:
may be we could use `RepoList` and take props `data`
```javascript
<RepoList data={favoriteList} />
<ReposList data={projects} />
```
so we could re-use `ReposList`

chucklien replied:
or 
```javascript
const FavList = <RepoList data={favoriteList} />;
const MyProject = <RepoList data={projects} />;
```
the purpose is used to DRY, maybe in the future you think they are not the same data structure, you could split easily

ritata replied:
thanks, I will refactor it.
use common component when display content and remove duplicate component.

ritata committed:
UPD. use common component when display Repositories content and remove duplicate component

ritata committed:
UPD. modify util to pure js function and remove duplicate component
1. modify ConvertToDate to pure js function
2. use common component when display content and remove duplicate component.
